src/redux/stores/metadata/metadataApi.ts

```diff
@@ -2,22 +2,31 @@ import { TokenInfo } from '@airswap/typescript';
 import { createAsyncThunk } from '@reduxjs/toolkit';
 import { ethers } from 'ethers';
 
-import { scrapeToken } from '../../../helpers/airswap';
+import { getTokenFromContract } from '../../../helpers/airswap';
 import { wait } from '../../../helpers/tools';
 
 interface ScrapeTokensParams {
-  tokens: string[];
-  library: ethers.providers.BaseProvider | null;
+  currencyToken: string;
+  collectionToken: string;
+  library: ethers.providers.BaseProvider;
   chainId?: number;
 }
 
-export const scrapeTokens = createAsyncThunk<(
+export const getCurrencyAndCollectionTokenInfo = createAsyncThunk<(
 TokenInfo | undefined)[], ScrapeTokensParams>(
-  'metadata/scrapeTokens',
-  async ({ tokens, library, chainId }) => Promise.all(tokens.map((token, index) => {
+  'metadata/getTokenFromContract',
+  async ({
+    currencyToken,
+    collectionToken,
+    library,
+    chainId,
+  }) => Promise.all([currencyToken, collectionToken].map((token, index) => {
     // On testnet we need to throttle calls or else we risk getting 429;
```

Comment:
This setup isn't throttling it's delaying

Reply:
> This setup isn't throttling it's delaying, I've seen 429's man_shrugging

![image](https://user-images.githubusercontent.com/12832288/218599916-ae2399ad-84a2-4cba-88f6-412d36416cc3.png)

Reply:
Screenshot is from the calls in `fetchCollectionTokens . So we need to throttle those too on testnet. Problem is testnet wasnt working for me last week. Maybe it works now with the new airswap library.

Reply:
What collection token are you using here?

Reply:
0xb46906699b80df34b7993afa40fb11470e51735b and scroll down the collections page fast :smile:

Reply:
Ok so this also happens on mainnet. We need to add some delay after loading. So I updated this ticket:

https://github.com/airswap/airswap-marketplace/issues/45